workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage.tsx

```diff
@@ -158,7 +137,11 @@ export function ReviewEndpointDiffPage(props: any) {
                 <DiffLinks
                   shapeDiffs={shapeDiffs.results}
                   setSelectedDiffHash={(hash: string) => {
-                    setSelectedDiffHash(hash);
+                    const hashIndex = shapeDiffs.results.findIndex(
```

Comment:
why is an index-based approach better than the hash-based approach here?

Reply:
So it's slightly simpler in this case - the interactions on this page are:
- on mount, we want to set the current index to 0 (which is possible undefined if there are no selected hashes)
- go forward or back in current index
- go to a specific hash

Using a current index, and then computing the selected diff is a simpler way to model the state (in either scenario we have to compute either or) and it makes the initial state setting simple (since we'll always start on currentIndex=0 instead of starting on the selectedDiffHash which is possibly undefined). There's isn't a great catchall reason here other than, it takes less code + less derivation to store index over selected hash

Ideally we don't compute new state + listen to state using useEffect (instead we use useMemo or useCallback to derive state)

Reply:
gotcha. yeah I didn't catch the original implementation in the diff, but as long as the assumptions of stability around the diff results holds then it is fine with me. Typically if I have a dropdown with next/prev, I'll set the next/prev when I render or change the current item so it's not computed on click